### PR TITLE
[common] major Configuration simplification

### PR DESCRIPTION
**Pull request by renemilk**
- strips away Request mapping and default recording
- removes functionally equivalent constructors and ``get`` variants
- test continue to pass (std::complex still failing) w/o modification
beyond removing calls to removed functionality.
- Esp. ``get`` calls should need no maintenance on the user side

### CODE REVIEW

**Comment by renemilk** [comment]
``DSC_CONFIG_GETV`` and ``DSC_CONFIG_GETB`` should be converted to functions too

File: dune/stuff/common/configuration.hh [hunk 1]
```diff
@@ -256,159 +180,61 @@ public:
...
-
   //! const get without default value, with validation
-  template< class T, class Validator >
-  T get(const std::string key,
-        const ValidatorInterface< T, Validator >& validator,
-        const size_t size = 0,
-        const size_t cols = 0) const
+  template< class T, class Validator = ValidateAny< typename internal::Typer< T >::type > >
+  typename internal::Typer<T>::type get(std::string key,
+        const size_t size,
```
Comment (ftalbrecht) [inline dune/stuff/common/configuration.hh]: Is the default value for `size` missing on purpose?

**Comment by ftalbrecht** [comment]
Wow, did not see that one coming. I never used the requests and have no objections against dropping them. The resulting simplifications look great! From a quick glance I could not estimate who would be calling which `get` method, though. Is the missing default value for `size` and the rearranged order of `size`, `cols` and `validator` on purpose/problematic? (Sorry for not investigating myself).

**Comment by renemilk** [comment]
It is on purpose and was necessary to be able to remove other signatures w/o getting ambiguous calls.  Basically anybody requesting vector/matrix like objects is putting in 1/2 size_t anyhow, so this does not impact user code. Only our test suite covers the use case  ofc, but those looked very detailed to me.

**Pull request merged**